### PR TITLE
CB-7608: Modify FreeIPA backup to check if azure backup location host resolves

### PR DESCRIPTION
**Pull request by wonderslug**
This modifies the FreeIPA backup script to check if the backup location 
host is valid. This is for azure only because the storage location 
is a customer provided hostname based on the storage account being used.

Closes #8365

### CODE REVIEW

**Comment by keyki** [comment]
I usually run shellcheck on the scripts, there are useful tips to make it more robust. 
```
Line 5:
  proto="$(echo $1 | grep :// | sed -e's,^\(.*://\).*,\1,g')"
                ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
  proto="$(echo "$1" | grep :// | sed -e's,^\(.*://\).*,\1,g')"
 
Line 7:
  url="$(echo ${1/$proto/})"
       ^-- SC2116: Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.
              ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
  url="$(echo "${1/$proto/}")"
 
Line 9:
  userpass="$(echo $url | grep @ | cut -d@ -f1)"
                   ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
  userpass="$(echo "$url" | grep @ | cut -d@ -f1)"
 
Line 10:
  pass="$(echo $userpass | grep : | cut -d: -f2)"
               ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
  pass="$(echo "$userpass" | grep : | cut -d: -f2)"
 
Line 12:
    user="$(echo $userpass | grep : | cut -d: -f1)"
                 ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
    user="$(echo "$userpass" | grep : | cut -d: -f1)"
 
Line 17:
  host="$(echo ${url/$user:$pass@/} | cut -d/ -f1)"
               ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
  host="$(echo "${url/$user:$pass@/}" | cut -d/ -f1)"
 
Line 18:
  echo $host
       ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
  echo "$host"
```

**Comment by wonderslug** [comment]
Updated with a lot of shell check cleanups

File: freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup [hunk 1]
```diff
@@ -93,25 +111,25 @@ elif [[ "$TYPE" = "DATA" ]]; then
...
 doStatus(){
     if [[ -n "$BACKUP_SCHEDULE" ]]
     then
-        type_of_msg=$(echo $*|cut -d" " -f1)
+        type_of_msg=$(echo "$*"|cut -d" " -f1)
         msg=$(echo "$*"|cut -d" " -f2-)
         [[ $type_of_msg == INFO ]] && type_of_msg="INFO " # one space for aligning
 
-        echo "`date "+%Y-%m-%dT%H:%M:%SZ"` $type_of_msg $msg" > $STATUSFILEPREFIX$BACKUP_SCHEDULE.log
+        echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg $msg" > "${STATUSFILEPREFIX}""${BACKUP_SCHEDULE}".log
```
Comment (jamisonbennett) [inline freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup:132]: Is there a reason to have double quotes nextto each other? `echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg $msg" > "${STATUSFILEPREFIX}${BACKUP_SCHEDULE}".log` might be a little cleaner.

File: freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup [hunk 2]
```diff
@@ -158,8 +176,12 @@ exlock_now || error_exit "A backup seems to be currently running. Lock file is a
...
 
 # Perform a backup
-/sbin/ipa-backup $BACKUP_OPTIONS >> $LOGFILE 2>&1 || error_exit "ipa-backup failed! Aborting!"
-BACKUPDIR=$(basename $(ls -td ${config[backup_path]}/ipa-* | head -1))
+
+# shellcheck disable=SC2086
+/sbin/ipa-backup ${BACKUP_OPTIONS} >> "${LOGFILE}" 2>&1 || error_exit "ipa-backup failed! Aborting!"
+
+# shellcheck disable=SC2012
+BACKUPDIR=$(basename "$(ls -td "${config[backup_path]}""/ipa-*" | head -1)")
```
Comment (jamisonbennett) [inline freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup:184]: Is there a reason the double quotes are back to back?
Wouldn't `BACKUPDIR=$(basename "$(ls -td "${config[backup_path]}/ipa-*" | head -1)")` be simpler?

File: freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup [hunk 3]
```diff
@@ -93,25 +111,25 @@ elif [[ "$TYPE" = "DATA" ]]; then
...
-    [[ $type_of_msg == DEBUG ]] && [[ $do_print_debug_msgs -ne 1 ]] && return
+    [[ $type_of_msg == DEBUG ]] && [[ ${PRINT_DEBUG_MSGS} -ne 1 ]] && return
     [[ $type_of_msg == INFO ]] && type_of_msg="INFO " # one space for aligning
     [[ $type_of_msg == WARN ]] && type_of_msg="WARN " # as well
 
     # print to the terminal if we have one
-    test -t 1 && echo "`date "+%Y-%m-%dT%H:%M:%SZ"` $type_of_msg ""$msg"
-    echo "`date "+%Y-%m-%dT%H:%M:%SZ"` $type_of_msg ""$msg" >> $LOGFILE
+    test -t 1 && echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg ""$msg"
+    echo "$(date "+%Y-%m-%dT%H:%M:%SZ") $type_of_msg ""$msg" >> "${LOGFILE}"
```
Comment (jamisonbennett) [inline freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup:122]: Same thing with double quoting here on these two lines.

Comment (wonderslug) [reply freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup]: fixed

File: freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup (hunk 2 above)
Comment (wonderslug) [reply freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup]: fixed

File: freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup (hunk 1 above)
Comment (wonderslug) [reply freeipa/src/main/resources/freeipa-salt/salt/freeipa/scripts/freeipa_backup]: fixed

**Pull request merged**